packages/prettier-plugin-java/src/printers/blocks-and-statements.ts

```diff
@@ -331,12 +335,25 @@ export class BlocksAndStatementPrettierVisitor extends BaseCstPrettierPrinter {
     } else {
       caseInstruction = concat([this.visit(ctx.expression), ctx.Semicolon![0]]);
     }
+    const caseLabelElement =
+      ctx.switchLabel[0].children.caseOrDefaultLabel[0].children
+        .caseLabelElement?.[0].children;
+    const recordPattern = caseLabelElement?.pattern?.[0].children.recordPattern;
+    const separator = recordPattern && !caseLabelElement?.guard ? " " : line;
 
-    return join(" ", [switchLabel, ctx.Arrow[0], caseInstruction]);
+    return concat([
+      group([switchLabel, separator, group([ctx.Arrow[0], line])]),
+      caseInstruction
+    ]);
   }
 
   caseConstant(ctx: CaseConstantCtx) {
-    return this.visitSingle(ctx);
+    const constant = this.visitSingle(ctx);
```

Comment:
I'm not sure why you need to do this ? Do you have a test sample to reproduce ?

Reply:
I think this was accidentally leftover from #606. However, something similar to this change seems necessary to properly indent `CaseConstant`s with parentheses (for example, a `BinaryExpression` wrapped with parentheses can be used as a `CaseConstant`). As I'm working on your suggestion in https://github.com/jhipster/prettier-java/pull/611#discussion_r1375387354, I'm trying to make a similar change for `CaseConstant`s (i.e. adding parentheses if they wrap).

Reply:
Never mind, this ended up being a lot more complicated than I thought. Since it's unrelated to the purpose of the PR (Java 21 pattern matching/guards support), I'm going to get rid of these changes and leave this improvement for another PR.